web/static/js/components/questionnaires/QuestionnaireTitle.jsx

```diff
@@ -30,18 +40,35 @@ class QuestionnaireTitle extends Component {
 
   render() {
     const { questionnaire, readOnly, t } = this.props
+    if (questionnaire) {
+      return (
+        <div className='title-container'>
+          <EditableTitleLabel title={questionnaire.name} onSubmit={(value) => { this.handleSubmit(value) }} emptyText={t('Untitled questionnaire')} readOnly={readOnly} more={<QuestionnaireMenu />} />
+          <EditableDescriptionLabel description={questionnaire.description} emptyText={readOnly ? '' : t('Add description')} onSubmit={(value) => { this.handleSubmitDescription(value) }} readOnly={readOnly} />
+        </div>
+      )
+    } else {
+      return null
+    }
+  }
+}
 
-    return (
-      <div className='title-container'>
-        <EditableTitleLabel title={questionnaire.name} onSubmit={(value) => { this.handleSubmit(value) }} emptyText={t('Untitled questionnaire')} readOnly={readOnly} more={<QuestionnaireMenu />} />
-        <EditableDescriptionLabel description={questionnaire.description} emptyText={t('Add description')} onSubmit={(value) => { this.handleSubmitDescription(value) }} readOnly={readOnly} />
-      </div>
+const mapStateToProps = (state, ownProps) => {
+  const { readOnly, params } = ownProps
+  const { projectId, questionnaireId, mode } = params
+  return {
+    projectId: parseInt(projectId),
+    questionnaireId: parseInt(questionnaireId),
+    mode,
+    questionnaire: state.questionnaire.data,
+    readOnly: typeof readOnly === 'boolean'
+    ? readOnly
+    : (
+      state.project && state.project.data
+      ? state.project.data.readOnly
+      : true
     )
   }
 }
 
-const mapStateToProps = (state, ownProps) => ({
-  readOnly: state.project && state.project.data ? state.project.data.readOnly : true
-})
-
-export default translate()(connect(mapStateToProps)(withQuestionnaire(QuestionnaireTitle)))
+export default translate()(withRouter(connect(mapStateToProps)(QuestionnaireTitle)))
```

Comment:
I'm checking that `withQuestionnaire` is responsible of putting the quex in redux store into de property of the component. Why this need to change? Other usages of `fetchQuestionnaireIfNeeded` reveals that the components referenced in the routes.jsx like `QuestionnaireEditor`and `SurveySimulation` are the ones responsibles to load from params into the redux if needed. Leaving the `QuestionnaireTitle` to grab the quex via `withQuestionnaire`.

Is there any issue to move `fetchQuestionnaireIfNeeded` into the `QuestionnaireSimulation`?

Reply:
Avoid fetching the questionnaire from `QuestionnaireSimulation` was done intentionally [here](https://github.com/instedd/surveda/pull/1698/commits/40b62e16f91fc789f5458a87bcfc0bb2b2559922).
The intention was to avoid fetching something that the component doesn't need for itself.
But I can revert that with no problem if you think it was better that way.

Reply:
I think is best to keep the same design: allow the top-level component to do the fetch from the url. Otherwise the components like title end up depending on the url, instead of receiving props.

Reply:
OK, so I'll revert that change. 👍

Reply:
Done!